src/components/shapes/attributes.js

```diff
@@ -224,6 +225,52 @@ module.exports = templatedArray('shape', {
             '`config.editable` or `config.edits.shapePosition`.'
         ].join(' ')
     },
-
+    label: {
+        text: {
+            valType: 'string',
+            dflt: '',
+            editType: 'arraydraw',
+            description: 'Sets the text to display with shape.'
+        },
+        font: fontAttrs({
+            editType: 'calc+arraydraw',
+            colorEditType: 'arraydraw',
+            description: 'Sets the shape label text font.'
+        }),
+        textposition: {
+            valType: 'enumerated',
+            values: [
+                'top left', 'top center', 'top right',
+                'middle left', 'middle center', 'middle right',
+                'bottom left', 'bottom center', 'bottom right',
+                'start', 'middle', 'end',
+            ],
+            editType: 'arraydraw',
+            description: 'Sets the position of the label text relative to the shape.'
+        },
+        textangle: {
+            valType: 'angle',
+            dflt: 0,
+            editType: 'calc+arraydraw',
+            description: [
+                'Sets the angle at which the `text` is drawn',
+                'with respect to the horizontal.'
```

Comment:
Describe the `'auto'` behavior and how it applies only to lines.
Also remove the `dflt` since you're setting it dynamically - and describe that logic here.